src/cljam/io/wig.clj

```diff
@@ -0,0 +1,180 @@
+                                   [start value] fields
+                                   start (as-long start)
+                                   end (dec (+ start span))
+                                   value (str->wiggle-track-data value)]
+                               {:format :variable-step, :chrom chrom, :start start, :end end, :value value})
+
+                             :fixed-step
+                             (let [{:keys [chrom pre-start span step]} current-track
+                                   start (+ pre-start step)
+                                   end (dec (+ start span))
+                                   value (-> fields first str->wiggle-track-data)]
+                               {:format :fixed-step, :chrom chrom, :start start, :end end, :value value})
+
+                             (throw (IllegalArgumentException. "Invalid wiggle format")))]
+                     (cons m (deserialize (rest lines)
+                                          (update current-track :pre-start (fn [_] (:start m)))))))))))]
+    (deserialize lines {:format nil, :chrom nil, :pre-start nil, :span nil :step nil})))
+
+(defn- serialize-wigs
+  "Serialize a sequence of WIG fields into a lazy sequence of string."
+  [wigs]
+  {:pre [;; These fields are required.
+         (map (fn [wig] (every? some? ((apply juxt wig-fields) wig))) wigs)
+         ;; There are two options for formatting wiggle data.
+         (map (fn [{:keys [format]} wig] (or (= format :fixed-step) (= format :variable-step))) wigs)]}
+  (letfn [(same-track? [first-wig wig]
+            (and (= (:format first-wig) (:format wig))
+                 (= (- (:end first-wig) (:start first-wig))
+                    (- (:end wig) (:start wig)))))
+          (serialize [wigs]
+            (lazy-seq
+             (when (some? (first wigs))
+               (let [[track rest-wigs] (split-with (partial same-track? (first wigs)) wigs)
+                     track (case (-> track first :format)
+                             :variable-step
+                             (let [{:keys [chrom start end]} (first track)
+                                   span (inc (- end start))
+                                   declaration-line (->> (cond-> ["variableStep" (str "chrom=" chrom)]
+                                                           (not= span 1) (conj (str "span=" span)))
+                                                         (cstr/join \space))
+                                   data-lines (->> track
+                                                   (map (fn [{:keys [start value]}] (cstr/join \space [start value])))
+                                                   (cstr/join \newline))]
+                               (cstr/join \newline [declaration-line data-lines]))
+
+                             :fixed-step
+                             (let [{:keys [chrom start end]} (first track)
+                                   step (- (-> track rest first :start) start)
```

Comment:
You can replace `(first (rest xs))` with `(second xs)`.

By the way, I think it would be great it could handle fixed-step tracks with a single data line, for just in case. The current implementation will throw an error in this line if the writer gets fed with such data.